Clean up and CONTRIBUTING upgrade (#595)
devicons/devicon: CONTRIBUTING.md

<h2 id="overview">Overview on Submitting Icons</h2>
<p>Here is what you have to do to submit your icons to the repo.</p>
<ol>

>   <li>Create the svgs for each <a href="#versionNaming"> svg versions </a> that you have. Follow the <a href="#svgStandards">convention</a> listed.</li>
>   <li>Put the svgs of each Icon into its own <a href="#orgGuidelines">folders</a> in <code>/icons</code></li>

Reviewer: ```suggestion
  <li>Create the SVGs for each <a href="#versionNaming"> SVG versions </a> that you have. Follow the <a href="#svgStandards">convention</a> listed.</li>
  <li>Put the SVGs of each Icon into its own <a href="#orgGuidelines">folders</a> in <code>/icons</code></li>
```

  <li><a href="#updateDevicon">Update the <code>devicon.json</code> to include the new Icon</a> </li>
  <li>Create a separated pull request (PR) towards the <code>develop</code> branch for each Icon.</li>
  <li>Include the name of the Icon in the pull request title in this format: <code>new icon: <i>Icon name</i> (<i>versions</i>)</code> </li>
  <li><i>Optional</i>: Reference the issues regarding the new icon. </li>
  <li>Some bots will check your SVGs. If there are any errors, please fix them as instructed.</li>
  <li>Wait for a maintainer to review your changes. They will run the <a href='#peekBot'><code>peek-bot</code></a> to check your icons.</li>

>   <li>If there are no issues, they will run the <a href='#optimizeBot'><code>optimize-bot</code></a> on your svgs then merge it using <a href="https://github.com/devicons/devicon/discussions/470">squash merging</a>. If there are any problems, they will let you know and give you a chance to fix it.</li>

Reviewer: ```suggestion
  <li>If there are no issues, they will run the <a href='#optimizeBot'><code>optimize-bot</code></a> on your SVGs then merge it using <a href="https://github.com/devicons/devicon/discussions/470">squash merging</a>. If there are any problems, they will let you know and give you a chance to fix it.</li>
```

  <li>Amazon Web Services becomes <code>amazonwebservices</code></li>
  <li>Microsoft SQL Server becomes <code>microsoftsqlserver</code></li>
</ul>
<br>

> <p>Each icon/svg can come in different versions:</p>

Reviewer: ```suggestion
<p>Each icon/SVG can come in different versions:</p>
```

  </li>

  <li style='margin-top: 3em'><b>plain-wordmark</b>: a one-color version of the original logo but with wordmark. Note that the icon version will be stripped of all colors so you don't have to strip beforehand. Example: <br>
  <img src='https://raw.githubusercontent.com/devicons/devicon/master/icons/gimp/gimp-plain-wordmark.svg' height='100px' width='100px' /> <code>gimp-plain-wordmark.svg</code>
  </li>

  <li style='margin-top: 3em'><b>line</b>: a one-color, line version of the original logo. Note that the icon version will be stripped of all colors so you don't have to strip beforehand. Example: <br>
  <img src='https://raw.githubusercontent.com/devicons/devicon/master/icons/apache/apache-line.svg' height='100px' width='100px' /> <code>apache-line.svg</code>
  </li>

  <li style='margin-top: 3em'><b>line-wordmark</b>: a one-color, line version of the original logo but with wordmark. Note that the icon version will be stripped of all colors so you don't have to strip beforehand. Example: <br>
  <img src='https://raw.githubusercontent.com/devicons/devicon/master/icons/apache/apache-line-wordmark.svg' height='100px' width='100px' /> <code>apache-line-wordmark.svg</code>

>   </li>

Reviewer 2: I appreciate the idea of example icons! Maybe we can use the devicon-icon as example instead of gimp and apache? We have it in all versions so I think it would be a good example. Using gimp and apache might be confusing for new contributors.

I was also wondering if a `table` instead of a unordered list (`<ul>`) would look cleaner for showing the examples?

Author: The table sounds like a good idea. Let me try it out.

Author: Hey @amacado, we unfortunately do not have `line` and `line-wordmark` versions of the `devicon` icons. What should I use for those ones instead?

Reviewer 2: I don't know yet.. Maybe we should create a issue to come up with a `line` and `line-wordmark` version of devicon.

    The <b>plain</b> and <b>line</b> versions (with or without wordmark) are designed to be available in the final icon font.
  </li>
  <li>

>     The <b>original</b> svg version do not need to be simple and can contain numerous colors/gradients. However, if it's intended to be made into an icon, keep it simple.

Reviewer: ```suggestion
    The <b>original</b> SVG version do not need to be simple and can contain numerous colors/gradients. However, if it's intended to be made into an icon, keep it simple.
```

<p>Before you submit your logos/svgs, please ensure that they meet the following standard:</p>
<ul>
  <li>The background must be transparent.</li>

>   <li>The icon is centered within the file.</li>

Reviewer 2: ```suggestion
  <li>The icon is centered horizontally and vertically within the <code>viewBox</code>.</li>
```

<p>Before you submit your logos/svgs, please ensure that they meet the following standard:</p>
<ul>
  <li>The background must be transparent.</li>
  <li>The icon is centered horizontally and vertically within the <code>viewBox</code>.</li>
  <li>The svg name follows this convention: <code>(Technology name)-(original|plain|line)(-wordmark?).</code></li>
  <li>The <b>plain</b> and <b>line</b> versions (with or without wordmark) need to stay as simple as possible. They must have only one color and the paths are united. The color will be removed when being turned into icons so the <code>.svg</code> can have any color.
  </li>

>   <li>Each <code>.svg</code> file contains one version of an icon in a <code>0 0 128 128</code> viewbox. You can use a service like <a href="https://www.iloveimg.com/resize-image/resize-svg">resize-image</a> for scaling the svg.</li>

Reviewer: ```suggestion
  <li>Each <code>.svg</code> file contains one version of an icon in a <code>0 0 128 128</code> viewbox. You can use a service like <a href="https://www.iloveimg.com/resize-image/resize-svg">resize-image</a> for scaling the SVG.</li>
```

  <li>The icon's strokes and texts must be fills. This is to satisfy our conversion website's <a href="https://icomoon.io/#docs/stroke-to-fill">requirements.</a></li>
  <li>The <code>svg</code> element does not need the <code>height</code> and <code>width</code> attributes. However, if you do use it, ensure their values are either <code>"128"</code> or <code>"128px"</code>. Ex: <code>height="128"</code></li>
  <li>Optimize/compress your SVGs. You can use a service like <a href="https://compressor.io/">compressor</a> or <a href="https://petercollingridge.appspot.com/svg-editor">SVG Editor</a>.</li>

>   <li>Each <code>.svg</code> must use the <code>fill</code> attribute instead of using <code>classes</code> for colors. This is to prevent class name clashing when using inline svg. See <a href="https://github.com/devicons/devicon/issues/407">here</a> for more details.</li>

Reviewer: ```suggestion
  <li>Each <code>.svg</code> must use the <code>fill</code> attribute instead of using <code>classes</code> for colors. This is to prevent class name clashing when using inline SVG. See <a href="https://github.com/devicons/devicon/issues/407">here</a> for more details.</li>
```

  </li>
  <li>Create a separate pull request (PR) for each Icon.
    <ul>

>       <li>This means you would have to create two PRs for each Icon.</li>

Reviewer 2: Nope. You don't have to create two pull requests for each icon. I think the version before is right. Or am I missing something? :smile:

Author: For this, I was referring to two different sets of icons. In the examples, I was submitting AWS and React to demonstrate when people want to tackle multiple technology. I think I'll rephrase it so it's clearer

<hr>
<h2 id='buildScript'>The Build Script: how it works and its quirks</h2>
<p>We rely on GitHub Actions, Python, Selenium, Imgur, and Gulp to automate our tasks. Please feel free to take a look at the workflow files. The codes should be clear enough to follow along.</p>

<p>Here are the main bots/script that we use:</p>
<ul>
  <li id='peekBot'><code>peek-bot</code>: Upload the icons to Icomoon and see what it looks like. Doesn't download any icons at this time.</li>
  <li id='buildBot'><code>build-bot</code>: Build the icons by uploading them to Icomoon and download the resulting icon files. Also update the css file</li>

>   <li id='optimizeBot'><code>optimize-bot</code>: Optimize the svgs by minifying them and prefixing their ids with the file names. This is done so using inline svgs from this repo will not cause id clash.</li>

Reviewer: ```suggestion
  <li id='optimizeBot'><code>optimize-bot</code>: Optimize the SVGs by minifying them and prefixing their IDs with the file names. This is done so using inline SVGs from this repository will not cause ID clash.</li>
```
